File: 백준/Gold/1107. 리모컨/리모컨.cc
#include<iostream>
#include<string>
#define FAST_IO cin.tie(0), cout.tie(0), ios_base::sync_with_stdio(0)
using namespace std;

int n, m, ans;
bool error[10];

void travel() {

	for (int i = 0; i < 1000000; i++) {
		string cur = to_string(i);

		bool is_error = false;
		for (auto e : cur) {

			if (error[e - '0']) {
				is_error = true;
				break;
			}
		}
		if (!is_error) {
			ans = min(ans, (int)cur.size() + abs(i - n));
		}
	}
}

int main(void) {

	FAST_IO;
	
	cin >> n >> m;

	for (int i = 0; i < m; i++) {
		int inp;
		cin >> inp;
		error[inp] = true;
	}

	ans = abs(n - 100);
	travel();

	cout << ans;

	return 0;


}

File: 백준/Gold/1107. 리모컨/리모컨.py
import sys

n = int(sys.stdin.readline())
m = int(sys.stdin.readline())

broken = set(sys.stdin.readline().split()) if m > 0 else set()

ans = abs(n - 100)

for channel in range(1000001):
    cur = str(channel)

    for char in cur:
        if char in broken:
            break
    else:
        ans = min(ans, len(cur) + abs(channel - n))

print(ans)
